### PR TITLE
[swift5] Fix target SDKs for Combine option

### LINKED ISSUE #7728: [BUG][SWIFT5] Swift5 generator sets wrong platforms requirement in Package.swift
##### Description

Swift5 generates `Package.swift` a manifest file for Swift Package Manger. The manifest file contains a `platforms` directive that specifies the minimum iOS, macOS, tvOS, watchOS SDK version required to compile the generaged sources. Swift5 generator config options include a `responseAs` configuration that specifies what libraries the generated code should use. When `responseAs` is set to `Combine`, the generated code will use the Combine Framework, which is available on iOS 13, tvOS 13, macOS 10.15, watchOS 6 and above. The `platforms` directive in the `Package.swift` file does not reflect this requirement.

###### Expected `Package.swift`:

```
platforms: [
        .iOS(.v13),
        .macOS(.v10_11),
        .tvOS(.v9),
        .watchOS(.v3)
    ],
```

###### Actual `Package.swift`:
```
platforms: [
        .iOS(.v9),
        .macOS(.v10_11),
        .tvOS(.v9),
        .watchOS(.v3)
    ],
```

##### openapi-generator version

4.3.1

##### OpenAPI declaration file content or url

Gradle.build.kts:
```kotlin
openApiGenerate {
    validateSpec.set(true)
    generatorName.set("swift5")
    outputDir.set(openApiOutputDir.toString())
    inputSpec.set(file("openapi.yaml").toString())
    configOptions.set(mapOf(
            "nonPublicApi" to "false",
            "responseAs" to "Combine",
            "projectName" to "SpeedtestPlus",
            "swiftUseApiNamespace" to "false"
    ))
}
```

##### Steps to reproduce

Create a Gradle build file with the above configuration, use any openapi.yaml, and generate.

##### Suggest a fix

When `responseAs` is set to `Combine`, the `platforms` directive in `Package.swift` should be as follows:

```
platforms: [
        .iOS(.v13),
        .macOS(.v10_15),
        .tvOS(.v13),
        .watchOS(.v6)
    ],
```

### PR DESCRIPTION
**Pull request by fl034**
Fixes #7728 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by fl034** [comment]
@jgavris (2017/07) @ehyche (2017/08) @Edubits (2017/09) @jaz-ah (2017/09) @4brunu (2019/11)

**Comment by 4brunu** [comment]
Hi,

First of all, thanks for taking the time to create this PR. 🙂

Could you please describe your use case on why to make this change?

I would like to discuss this PR.

The methods that use `Combine` are already marked as `@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)`.
So this should already protect agains users of iOS 12 and bellow.

Is this not enough?

It should be possible to choose more than one `responseAs` at the same time, for example `Result` and `Combine`.
In this case, the minimum supported version of iOS should be 9, although it still uses `Combine`.

The current setup of mark the `Combine` methods with `@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)` should allow the users of the library to choose `Combine` in iOS 13 and above, and `Result` on iOS 12 and bellow.

One thing that maybe we could do is to change the minimum iOS version to 13 when `Combine` is the only selected `responseAs` option?

@fl034 What do you think about this?

**Comment by 4brunu** [comment]
Hi @fl034,

Looks like a good solution to me.

Are you thinking in adding `#canImport Combine` in the `import Combine` and also the methods that use `Combine?

**Comment by 4brunu** [comment]
Hi @fl034,

Thanks for making the changes that we talked about 👍 

I think that we should also wrap the methods that use `Combine` with  `#canImport Combine`.

What do you think about that?

**Comment by fl034** [comment]
Man you're fast! I was just writing about it. 
I was thinking to replace the `@available` if `#canImport`, but this leads to errors.
And leaving it as it is and only wrapping the import statement also leads to errors.
So you're right, we need both.
I will update it soon, again.

File: modules/openapi-generator/src/main/resources/swift5/api.mustache
Comment (4brunu) [inline modules/openapi-generator/src/main/resources/swift5/api.mustache:14]: Could you please put `#endif{{/useCombine}}{{#swiftUseApiNamespace}}` in the same line to avoid generating an extra line?

**Review by 4brunu — APPROVED** [review]
Looks good, I just left a small comment. 👍 
Thanks for your help!

File: modules/openapi-generator/src/main/resources/swift5/api.mustache
Comment (fl034) [reply modules/openapi-generator/src/main/resources/swift5/api.mustache:14]: Jep :-)

**Comment by fl034** [comment]
@4brunu I noticed that in the latest snapshot version, using URLSession as library, I don't get callbacks from my requests. It works with Alamofire as library. Also it works in 5.0.0 stable. 
Do you know if there's already an issue for that?

**Comment by 4brunu** [comment]
No, I didn't knew about that.
Do you have a sample project to show the issue?
Because I'm using the snapshot version with URLSession and it's working.

**Comment by fl034** [comment]
@4brunu I opened an issue #8511 regarding my observation.
Since it's unrelated to this PR, you can merge this PR if you want :-).

**Comment by 4brunu** [comment]
I don't have permissions to merge the PRs.
I only help reviewing them. 🙂 
Let's just wait for the CI to finish.

**Pull request merged**
